src/Dapr.Workflow/DaprWorkflowClientBuilderFactory.cs

```diff
@@ -28,27 +28,25 @@ namespace Dapr.Workflow;
 /// </summary>
 internal sealed class DaprWorkflowClientBuilderFactory
 {
-    private readonly IConfiguration _configuration;
+    private readonly IConfiguration? _configuration;
     private readonly IHttpClientFactory _httpClientFactory;
-    private readonly IServiceCollection _services;
     
     /// <summary>
     /// Constructor used to inject the required types into the factory.
     /// </summary>
-    public DaprWorkflowClientBuilderFactory(IConfiguration configuration, IHttpClientFactory httpClientFactory, IServiceCollection services)
+    public DaprWorkflowClientBuilderFactory(IConfiguration? configuration, IHttpClientFactory httpClientFactory)
     {
         _configuration = configuration;
         _httpClientFactory = httpClientFactory;
-        _services = services;
     }
     
     /// <summary>
     /// Responsible for building the client itself.
     /// </summary>
     /// <returns></returns>
-    public void CreateClientBuilder(Action<WorkflowRuntimeOptions> configure)
+    public void CreateClientBuilder(IServiceCollection services, Action<WorkflowRuntimeOptions> configure)
```

Comment:
It's not obvious to me why `DaprWorkflowClientBuilderFactory` itself needs to be registered with the container as opposed to just making this `CreateClientBuilder()` method an `IServiceCollection` extension method.  Is it because `AddDurabletaskClient()` needs a registered `IHttpClientFactory` and `IConfiguration`?  If so, you *should* be able to get them using the `builder.Services`.

Reply:
You're right - that was an artifact of having put together the fix that I neglected to remove. I've removed that registration and am resolving the services myself to create a new instance within the service provider scope.